packages/tasit-act/src/TasitAct.test.js

```diff
@@ -0,0 +1,137 @@
+import { Contract } from "./TasitAct";
+import Account from "tasit-account";
+import { expect, assert } from "chai";
+import { waitForEvent, mineBlocks } from "./helpers/utils.js";
+
+// Note: SimpleStorage.json is originally genarated by `tasit-contracts` and was pasted here manually
+// See https://github.com/tasitlabs/TasitSDK/issues/45
+import { abi as contractABI } from "./helpers/SimpleStorage.json";
+
+// Note: Under the current `tasit-contracts` setup SimpleStorage aways will deployed with this address
+// See https://github.com/tasitlabs/TasitSDK/pull/59#discussion_r242258739
+const contractAddress = "0x6C4A015797DDDd87866451914eCe1e8b19261931";
+
+describe("Contract", function() {
+  let simpleStorage, wallet;
+
+  beforeEach("should connect to an existing contract", async () => {
+    // Account creates a wallet, should it create an account object that encapsulate the wallet?
+    // TasitAcount.create()
+    // > Acount { wallet: ..., metaTxInfos..., etc }
+    wallet = Account.createFromPrivateKey(
+      "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60"
+    );
+
+    simpleStorage = new Contract(contractAddress, contractABI);
+    expect(simpleStorage).to.exist;
+    expect(simpleStorage.address).to.equal(contractAddress);
+    expect(simpleStorage.getValue).to.exist;
+    expect(simpleStorage.setValue).to.exist;
+    // Events are not implemented yet
+    //expect(simpleStorage.ValueChanged).to.exist;
+  });
+
+  it("should throw error when instantiated with invalid args", async () => {
+    try {
+      new Contract();
+      assert(false, "constructor without address and ABI");
+
+      new Contract(contractAddress);
+      assert(false, "constructor without ABI");
+
+      new Contract("invalid address");
+      assert(false, "constructor with invalid address");
```

Comment:
Is it resolved after lasts PRs?

Reply:
I don't think so. I think that this:

`constructor with invalid address and valid ABI`
https://github.com/tasitlabs/TasitSDK/blob/4fe56d3519d5c108f1b10d23796225d188dbbff3/packages/tasit-action/src/TasitAction.test.js#L109

makes this test:

`constructor with invalid address`
https://github.com/tasitlabs/TasitSDK/blob/4fe56d3519d5c108f1b10d23796225d188dbbff3/packages/tasit-action/src/TasitAction.test.js#L103

unnecessary.

Reply:
Also, we could add a test for "constructor without address but with ABI where the first arg is null, because that doesn't exist yet and isn't really covered fully. Feel free to extract this as a good first issue.

Reply:
Got it, thanks! PR: https://github.com/tasitlabs/TasitSDK/pull/139